src/System/include/BipedalLocomotion/System/SharedResource.h

```diff
@@ -44,6 +46,21 @@ template <class T> class SharedResource
      */
     T get();
 
+    /**
+     * Get the resource using const reference.
+     * @note one must not forget to access the mutex and
+     *       perform necessary locking and unlocking
+     * @return the const reference of the object inside the shared resource.
+     */
+    const T& get() const;
+
+    /**
+     * Get the mutex by reference.
+     * @note one must be careful not to modify the mutex
```

Comment:
> std::mutex is neither copyable nor movable.
From https://en.cppreference.com/w/cpp/thread/mutex

So in principle, there is no way to edit it, right?

Reply:
Great! Then, I will remove that line.

Reply:
Thanks, done in https://github.com/dic-iit/bipedal-locomotion-framework/pull/398/commits/7959a8d53b630076205238f2a2e5917366a5380a.